src/com/trilead/ssh2/crypto/digest/MAC.java

```diff
@@ -1,68 +1,114 @@
-
 package com.trilead.ssh2.crypto.digest;
 
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Mac;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.SecretKeySpec;
+
 /**
  * MAC.
  * 
  * @author Christian Plattner, plattner@trilead.com
  * @version $Id: MAC.java,v 1.1 2007/10/15 12:49:57 cplattne Exp $
  */
-public final class MAC
+public final class MAC 
 {
-	Digest mac;
-	int size;
```

Comment:
Similar to other comments about field visibility and backwards compatibility - I'd probably create a new version of this class and migrate any code that uses the current version.